docs/guides/integration-javascript.md

```diff
@@ -0,0 +1,293 @@
+
+UID2 provides a UID2 SDK for JavaScript (TODO add links!!!) with the following features:
+
+- UID2 token generation
+- Automatic refreshing of UID2 tokens
+- Automatic storage of UID2 tokens in the browser
+
+You'll need to complete the following steps:
+
+1. [Complete UID2 account setup](#complete-uid2-account-setup)
+2. [Add UID2 SDK For JavaScript to Your Site](#add-js-to-your-site)
+3. [Configure the UID2 SDK For JavaScript](#configure-the-uid2-sdk-for-javascript)
+4. [Retrieving the UID2 Advertising Token](#retrieving-the-uid2-advertising-token)
+
+
+## UID2 SDK for JavaScript Version
+
+Support for client-side token generation is available in version 3.2 and above of the SDK. 
+
+The URL for the SDK is https://cdn.prod.uidapi.com/uid2-sdk-3.2.0.js . In the following code examples, we will use the placeholder ```{{ UID2_JS_SDK_URL }}``` to refer to this URL.
+
+If you want to use a debug build of the SDK, use the following URL instead: https://cdn.integ.uidapi.com/uid2-sdk-3.2.0.js . 
+
+## Sample Implementation Website
+
+For an example application, see this example:
+- Running site: [UID2 SDK Integration Example](https://cstg-integ.uidapi.com/)
+
+## Complete UID2 Account Setup
+
+Complete the UID2 account setup by following the steps described in the [Account Setup](../getting-started/gs-account-setup.md) page. As part of the account setup process, you'll need to provide a list of domain names for the sites that you'll be using with this UID2 SDK for JavaScript.
+
+When account setup is complete, you'll receive a **public key** and **subscription ID**. These values are unique to you, and you'll use them to configure the UID2 module.
+
+:::tip
+Only root-level domains are required for account setup. For example, if you're going to use UID2 SDK for JavaScript on example.com, shop.example.com, and example.org, you only need to provide the domain names example.com and example.org.
+:::
+
+
+## Add UID2 SDK For JavaScript to Your Site
+
+The following code illustrates the different SDK events that we will respond to.
+
+```js
+<script defer src="{{ UID2_JS_SDK_URL }}"></script>
+ 
+<script>
+ 
+// When the UID2 SDK is executed, it will look for these callbacks and invoke them.
+window.__uid2 = window.__uid2 || {};
+window.__uid2.callbacks = window.__uid2.callbacks || [];
+window.__uid2.callbacks.push((eventType, payload) => {
+  switch (eventType) {
+    case "SdkLoaded":
+      // The SdkLoaded event occurs just once.
+      __uid2.init({});
+      break;
+ 
+    case "InitCompleted":
+      // The InitCompleted event occurs just once.
+      //
+      // If there was a valid UID2 token, it will be in payload.identity.
+      break;
+ 
+    case "IdentityUpdated":
+      // The IdentityUpdated event will happen when a UID2 token was generated or refreshed.
```

Comment:
I think it's worth adding that payload.identity will contain a valid UID2 token if one is available for this event too.